src/cycles/eulerian.jl

```diff
@@ -0,0 +1,318 @@
+# Checking for existence of Eulerian circuit and Eulerian trail
+
+function single_nonzero_degree_component(
+    graph::AG,
+    ) where {T, AG <: AbstractGraph{T}}
+
+    visited = zeros(UInt8, nv(graph))
+    for v in vertices(graph)
+        if (outdegree(graph, v) + indegree(graph, v)) == 0
+            visited[v] = 2
+        end
+    end
+    for v in vertices(graph)
+        if (outdegree(graph, v) + indegree(graph, v)) == 0
+            continue
+        end
+        S = Vector{T}([v])
```

Comment:
`S` is a stack? In that case it would be better to call it something like `stack`, so the code is easier to understand.